tests/integration/testlib/statelib.py

```diff
@@ -0,0 +1,53 @@
+#
+# Copyright 2018 Red Hat, Inc.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+#
+
+from libnmstate import netinfo
+
+
+INTERFACES = 'interfaces'
+
+
+def show_only(ifnames):
+    """
+    Report the current state, filtering based on the given interface names.
+    """
+    base_filter_state = {
+        INTERFACES: [{'name': ifname} for ifname in ifnames]
+    }
+    return filter_current_state(base_filter_state)
+
+
+def filter_current_state(desired_state):
```

Comment:
Maybe use a name like `get_corresponding_current_state` - it is strange that something that filters the current state gets a desired state as argumentr. Or maybe make the current state an argument as well and use a matching name. Since it get's the current state in the function I would even argue that this should be a class method for a class that represents the current state - but since it is not a class, it is a strange method.